src/main/java/com/microsoft/jenkins/azuread/AzureAuthenticationToken.java

```diff
@@ -55,7 +55,14 @@ public void setAuthenticated(boolean isAuthenticated) throws IllegalArgumentExce
 
     @Override
     public String getName() {
-        return (azureAdUser != null ? azureAdUser.getObjectID() : null);
+        return (azureAdUser != null ? getShortUpn(azureAdUser.getUsername()) : null);
+    }
+
+    private String getShortUpn(String username) {
+        if (username.contains("@")) {
```

Comment:
from issue description, user would like to fill in something without domain name `@xx`. 
```
It would be great to be able to use upn without the @xxxxxxx.xxx part too.
```

Reply:
Yes, I remove the @xx part by substring method in the below line.

Reply:
i  think user's ask is that they don't want to input the domain part.

Reply:
I suggest preserving the raw UPN since the prefix part could be same for different users which is problematic to be an id.

Besides, UPN is more than an email address. For example, `live.com#abc@gmail.com` is a valid UPN and is different from `abc@gmail.com`.